### PR TITLE
Feature/plotting cosmetic

### PR DESCRIPTION
**Pull request by martinfalk**

### CODE REVIEW

**Review by petersteneteg — DISMISSED** [review]
LGTM

File: modules/plottinggl/src/plotters/scatterplotgl.cpp [hunk 1]
```diff
@@ -614,32 +605,16 @@ void ScatterPlotGL::renderAxis(const size2_t& dims) {
...
 
     // Show tooltip for current item
-    if (rowIndex) {
-        if (p->getHoverState() == PickingHoverState::Move ||
-            p->getHoverState() == PickingHoverState::Enter) {
+    if (p->getHoverState() == PickingHoverState::Move ||
+        p->getHoverState() == PickingHoverState::Enter) {
 
-            tooltipCallback_.invoke(p, rowIndex.value());
+        tooltipCallback_.invoke(p, id);
```
Comment (ResearchDaniel) [inline modules/plottinggl/src/plotters/scatterplotgl.cpp:614]: How do you know that pickingId = dataframe row id?

**Review by petersteneteg — DISMISSED** [review]

File: modules/plottinggl/src/plotters/scatterplotgl.cpp [hunk 2]
```diff
@@ -409,19 +410,9 @@ void ScatterPlotGL::plot(const size2_t& dims, IndexBuffer* indexBuffer, bool use
...
-                    if (selected && !filtered) {
-                        selectedIndices.push_back(indexCol[ind]);
-                    }
-                }
-            } else {
-                for (auto [ind, selected, filtered] : util::enumerate(selected_, filtered_)) {
-                    if (selected && !filtered) {
-                        selectedIndices.push_back(static_cast<uint32_t>(ind));
-                    }
+            for (auto [ind, selected, filtered] : util::enumerate(selected_, filtered_)) {
```
Comment (ResearchDaniel) [inline modules/plottinggl/src/plotters/scatterplotgl.cpp:413]: brushing and linking contain the identifier (not the row)?

File: modules/plottinggl/src/processors/scatterplotprocessor.cpp [hunk 3]
```diff
@@ -198,7 +198,7 @@ void ScatterPlotProcessor::process() {
 
     BitSet b(brushingPort_.getFilteredIndices());
     // invert the bitset to obtain the ids remaining after the filtering
-    b.flipRange(0, static_cast<uint32_t>(dataframe->getNumberOfRows()));
+    b.flipRange(0, static_cast<uint32_t>(dataframe->getIndexColumn()->getRange().y));
```
Comment (ResearchDaniel) [inline modules/plottinggl/src/processors/scatterplotprocessor.cpp]: What about min range?

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Not sure if the new use of brushing id is correct

File: modules/plottinggl/src/plotters/scatterplotgl.cpp (hunk 2 above)
Comment (martinfalk) [reply modules/plottinggl/src/plotters/scatterplotgl.cpp:413]: ScatterPlotGL only uses zero-based indices, i.e. row numbers. The translation to IDs is handled by the ScatterPlot processor.

File: modules/plottinggl/src/processors/scatterplotprocessor.cpp (hunk 3 above)
Comment (martinfalk) [reply modules/plottinggl/src/processors/scatterplotprocessor.cpp]: Hm, could potentially use that one as well. However, there will hardly be any performance difference I think.

File: modules/plottinggl/src/plotters/scatterplotgl.cpp (hunk 1 above)
Comment (martinfalk) [reply modules/plottinggl/src/plotters/scatterplotgl.cpp:614]: row ids as well as picking ids are zero-based.
See https://github.com/inviwo/inviwo/blob/7fdcf8a8e65dc9c996f15e5c0649545171fcb7af/modules/plottinggl/src/processors/scatterplotprocessor.cpp#L77-L82
and 
https://github.com/inviwo/inviwo/blob/7fdcf8a8e65dc9c996f15e5c0649545171fcb7af/modules/dataframe/src/util/dataframeutil.cpp#L543-L554

File: modules/plottinggl/src/plotters/scatterplotgl.cpp (hunk 2 above)
Comment (martinfalk) [reply modules/plottinggl/src/plotters/scatterplotgl.cpp:413]: They always have. Compare for example with the PCP
https://github.com/inviwo/inviwo/blob/7fdcf8a8e65dc9c996f15e5c0649545171fcb7af/modules/plottinggl/src/processors/parallelcoordinates/parallelcoordinates.cpp#L490-L502

File: modules/plottinggl/src/plotters/scatterplotgl.cpp (hunk 1 above)
Comment (ResearchDaniel) [reply modules/plottinggl/src/plotters/scatterplotgl.cpp:614]: But this means that you cannot use brushing and linking for different data frames linked via their index column?
E.g., where the identifiers are the same, but the order is different

Comment (martinfalk) [reply modules/plottinggl/src/plotters/scatterplotgl.cpp:614]: On the contrary. `ScatterPlotGL` is just concerned with the actual plotting. Indices are provided from the "outside". The **processor** for a scatterplot performs this mapping. 

https://github.com/inviwo/inviwo/blob/62f9ea434b6a44f7ea76808aa7b108dc6829fe74/modules/plottinggl/src/processors/scatterplotprocessor.cpp#L97-L107

https://github.com/inviwo/inviwo/blob/62f9ea434b6a44f7ea76808aa7b108dc6829fe74/modules/plottinggl/src/processors/scatterplotprocessor.cpp#L180-L192

**Review by ResearchDaniel — DISMISSED** [review]

File: modules/dataframe/src/processors/dataframeexporter.cpp [hunk 4]
```diff
@@ -148,7 +148,7 @@ void DataFrameExporter::exportAsCSV(bool separateVectorTypesIntoColumns) {
         const auto components = col->getBuffer()->getDataFormat()->getComponents();
         if (components > 1 && separateVectorTypesIntoColumns) {
             for (size_t k = 0; k < components; k++) {
-                oj = citation + col->getHeader() + ' ' + componentNames[k] + citation;
+                oj = fmt::format("{0}{} {}{0}", citation, col->getHeader(), componentNames[k]);
```
Comment (petersteneteg) [inline modules/dataframe/src/processors/dataframeexporter.cpp:151]: Might add units too

Comment (petersteneteg) [reply modules/dataframe/src/processors/dataframeexporter.cpp:151]: not that we parse then though....

**Review by petersteneteg — APPROVED** [review]

File: modules/dataframe/src/processors/dataframeexporter.cpp (hunk 4 above)
Comment (martinfalk) [reply modules/dataframe/src/processors/dataframeexporter.cpp:151]: That's the thing. I would expect that a DataFrame which was exported to a file and then read again results in the same DataFrame, units in the header would change that. Unless we have support for parsing, as pointed out.
I'd move that to a future PR.

**Pull request merged**